components/camel-tahu/src/main/java/org/apache/camel/component/tahu/TahuEdgePayloadConverter.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.tahu;
+
+import java.util.Date;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+// import org.apache.camel.Converter;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.TypeConversionException;
+import org.apache.camel.TypeConverter;
+// import org.apache.camel.spi.HeaderFilterStrategy;
+// import org.apache.camel.spi.HeaderFilterStrategyAware;
+import org.apache.camel.spi.TypeConverterRegistry;
+import org.eclipse.tahu.SparkplugInvalidTypeException;
+import org.eclipse.tahu.message.model.Metric;
+import org.eclipse.tahu.message.model.MetricDataType;
+import org.eclipse.tahu.message.model.SparkplugBPayload;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// WIP
+// @Converter(generateLoader = true)
+public class TahuEdgePayloadConverter { // implements HeaderFilterStrategyAware {
```

Comment:
Note to myself: to-be-reviewed after it's ready.

Reply:
So of course, adding the PayloadConverter is causing legitimate TCK test failures. Will be getting those straightened out today and will let you know when it's ready for review.